snap/snap.go

```diff
@@ -62,19 +63,109 @@ func addPointsAndSnap(ix *PointIndex, polygon *geom.Polygon) *geom.Polygon {
 				newPolygon = append(newPolygon, newRing)
 			}
 		default:
-			// deduplicate points in the ring, then add to the polygon
-			newPolygon = append(newPolygon, kmpDeduplicate(newRing))
+			// deduplicate points in the ring, check winding order, then add to the polygon
+			deduplicatedRing := kmpDeduplicate(newRing)
+			// inner rings (ringIdx != 0) should be clockwise
+			shouldBeClockwise := ringIdx != 0
+			// winding order is reversed if incorrect
+			validateWindingOrder(deduplicatedRing, shouldBeClockwise)
+			newPolygon = append(newPolygon, deduplicatedRing)
 		}
 	}
 	return (*geom.Polygon)(&newPolygon)
 }
 
+// validate winding order (CCW for outer rings, CW for inner rings)
+// if winding order is incorrect, ring is reversed to correct winding order
+func validateWindingOrder(ring [][2]float64, shouldBeClockwise bool) {
```

Comment:
ik zou hem `ensureWindingOrder` noemen omdat hij niet alleen valideert maar ook corrigeert.

even dubbelcheck, die reverse gebeurt in place, en de ring is een slice en wordt dus niet als kopie maar byref meegegeven?

Reply:
Betere naam inderdaad, pas ik zo even aan. En de reverse is inderdaad in place, wat werkt omdat ik niet eerste een kopie van de ring maak